boxoffice/models/order.py

```diff
@@ -114,3 +114,26 @@ def is_fully_assigned(self):
             if not line_item.current_assignee:
                 return False
         return True
+
+
+class OrderSession(BaseMixin, db.Model):
+    """
+    Records the referrer and utm headers for an order
+    """
+    __tablename__ = 'order_session'
+    __uuid_primary_key__ = True
+
+    customer_order_id = db.Column(None, db.ForeignKey('customer_order.id'), nullable=False, index=True, unique=False)
+    order = db.relationship(Order, backref=db.backref('session', cascade='all, delete-orphan', uselist=False))
+
+    referrer = db.Column(db.Unicode(2083), nullable=True)
+
+    # Google Analytics parameters
+    utm_source = db.Column(db.Unicode(250), nullable=False, default=u'')
+    utm_medium = db.Column(db.Unicode(250), nullable=False, default=u'')
+    utm_term = db.Column(db.Unicode(250), nullable=False, default=u'')
+    utm_content = db.Column(db.Unicode(250), nullable=False, default=u'')
+    utm_id = db.Column(db.Unicode(250), nullable=False, default=u'')
+    utm_campaign = db.Column(db.Unicode(250), nullable=False, default=u'')
+    # Google click id (for AdWords)
+    gclid = db.Column(db.Unicode(250), nullable=False, default=u'')
```

Comment:
I suspect there will also be performance issues despite the index, as you'll want results scoped by orders that refer to an item collection, so it's a significant join. Let's deal with that later.